SW-639_update_readme (#203)
findologic/plugin-shopware-6: README.md

[![Build Status](https://github.com/findologic/plugin-shopware-6/workflows/PHPUnit/badge.svg?branch=main)](https://github.com/findologic/plugin-shopware-6/actions)

> ## Table of Contents

Author: Table of contents is automatically generated by GitHub since a few months, so we no longer need to add our own.

> 

Reviewer: ```suggestion
## Installation

### Production

Install the Findologic plugin from the [Shopware store](https://store.shopware.com/fin6494960032131f/findologic-suche-navigation.html).
```

> ## Development

Reviewer: ```suggestion
### Development
```

> ### Installation

Reviewer: ```suggestion
```

> These steps are only relevant for development.  

Reviewer: This should be clear when within development section.

> For an installation in production, please see [our Shopware 6 documentation](https://docs.findologic.com/doku.php?id=integration_documentation:plugin:en:integration:shopware_6).

Reviewer: Move this up to production section

1. Set up the [Shopware 6 development template](https://developer.shopware.com/docs/guides/installation/docker)
2. Navigate to the plugins folder
   ```bash

>    > cd custom/plugins

Reviewer: Remove the `>` because it's annoying when trying to copy.

   ```
3. Clone this repository
   ```bash
   > git clone git@github.com:findologic/plugin-shopware-6.git
   ```
4. Install dependencies
   ```bash
   > cd plugin-shopware-6 && composer install
   ```
5. Navigate back to the development template root
   ```bash
   > cd ../../..
   ```
6. Run Shopware if not already done
   ```bash
   > ./psh.phar docker:start
   ```
7. SSH into the Shopware docker container
   ```bash
   > ./psh.phar docker:ssh
   ```
8. Install and activate the plugin
   ```bash
   > bin/console plugin:install FinSearch --activate --clearCache
   ```

### Run PHPUnit tests

#### Prerequisites

* MySQL >= 8.0
* PHP >= 7.3

> #### Set up

Reviewer: ```suggestion
#### Setup
```

1. Create a database user `app` with the password `app`, which should have permissions to create,
   update and delete all databases.

> 2. In the development template's root initialize the test setup (this must be run locally, outside the app container)

Reviewer: ```suggestion
2. In the development templates, root initialize the test setup (Run locally, outside the app container)
```

   ```bash
   > ./psh.phar init
   ```
3. Navigate to the plugin and run tests
   ```bash
   > cd custom/plugins/plugin-shopware-6 && composer test
   ```

Tests can also be run separately within the IDE. Simply ensure to set `phpunit.xml.dist` as configuration file.

> ![PHPUnit configuration](docs/images/phpunit_configuration.png)

Author: This image seems to be broken, but once merged the link is correct.

### Developing custom JavaScript plugins

- Check out the
  [offical guide](https://docs.shopware.com/en/shopware-platform-dev-en/how-to/extend-core-js-storefront-plugin)
  on how to extend js storefront plugin.
- Create your plugin files inside
  `src/Resources/app/storefront/src/js/[your-plugin-name].plugin.js`
- [Register your extended plugin](https://docs.shopware.com/en/shopware-platform-dev-en/how-to/extend-core-js-storefront-plugin#register-your-extended-plugin)

> - For a development build, use `./psh.phar storefront:dev`

Reviewer: ```suggestion
- For development build: `./psh.phar storefront:dev`
```

> - For a production build, use `./psh.phar storefront:build`

Reviewer: ```suggestion
- For production build: `./psh.phar storefront:build`
```

**Please note:**

The build commands will create a minified JS file in `src/Resources/app/storefront/dist/storefront/js/[plugin-name].js`.
Before committing ensure that all files were built and added to your commit. Make sure to also commit the minified

> JavaScript files.

Reviewer: Maybe also use list points here?
